Build fails because implicit dependency of VerifyReleaseTask

with axion 14.4.3 and Gradle 8.1.1 I get the following error when building:

Reason: Task ':verifyRelease' uses this output of task ':app:compileJava' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.


<details open>
  <summary>Complete output</summary>

```
./gradlew test release                                                            

FAILURE: Build failed with an exception.

* What went wrong:
Some problems were found with the configuration of task ':verifyRelease' (type 'VerifyReleaseTask').
  - Gradle detected a problem with the following location: '/Users/x/Downloads/axiontest'.
    
    Reason: Task ':verifyRelease' uses this output of task ':app:compileJava' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':app:compileJava' as an input of ':verifyRelease'.
      2. Declare an explicit dependency on ':app:compileJava' from ':verifyRelease' using Task#dependsOn.
      3. Declare an explicit dependency on ':app:compileJava' from ':verifyRelease' using Task#mustRunAfter.
    
    Please refer to https://docs.gradle.org/8.1.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/x/Downloads/axiontest'.
    
    Reason: Task ':verifyRelease' uses this output of task ':app:compileTestJava' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':app:compileTestJava' as an input of ':verifyRelease'.
      2. Declare an explicit dependency on ':app:compileTestJava' from ':verifyRelease' using Task#dependsOn.
      3. Declare an explicit dependency on ':app:compileTestJava' from ':verifyRelease' using Task#mustRunAfter.
    
    Please refer to https://docs.gradle.org/8.1.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/x/Downloads/axiontest'.
    
    Reason: Task ':verifyRelease' uses this output of task ':app:test' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':app:test' as an input of ':verifyRelease'.────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
      2. Declare an explicit dependency on ':app:test' from ':verifyRelease' using Task#dependsOn.
      3. Declare an explicit dependency on ':app:test' from ':verifyRelease' using Task#mustRunAfter.
    
    Please refer to https://docs.gradle.org/8.1.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/x/Downloads/axiontest'.
    
    Reason: Task ':verifyRelease' uses this output of task ':app:processResources' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':app:processResources' as an input of ':verifyRelease'.
      2. Declare an explicit dependency on ':app:processResources' from ':verifyRelease' using Task#dependsOn.
      3. Declare an explicit dependency on ':app:processResources' from ':verifyRelease' using Task#mustRunAfter.
    
    Please refer to https://docs.gradle.org/8.1.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/x/Downloads/axiontest'.
    
    Reason: Task ':verifyRelease' uses this output of task ':app:processTestResources' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':app:processTestResources' as an input of ':verifyRelease'.
      2. Declare an explicit dependency on ':app:processTestResources' from ':verifyRelease' using Task#dependsOn.
      3. Declare an explicit dependency on ':app:processTestResources' from ':verifyRelease' using Task#mustRunAfter.
    
    Please refer to https://docs.gradle.org/8.1.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
```
</details>

That seems to be fixed in 1.15.4, but unfortunately we are stuck with Java 8.

--- a/src/main/groovy/pl/allegro/tech/build/axion/release/domain/RepositoryConfig.groovy
+++ b/src/main/groovy/pl/allegro/tech/build/axion/release/domain/RepositoryConfig.groovy
@@ -1,12 +1,10 @@
 package pl.allegro.tech.build.axion.release.domain
 
 import org.gradle.api.file.Directory
-import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.file.RegularFileProperty
 import org.gradle.api.provider.Property
 import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.Input
-import org.gradle.api.tasks.InputDirectory
 import org.gradle.api.tasks.InputFile
 import org.gradle.api.tasks.Optional
 
@@ -38,16 +36,16 @@ abstract class RepositoryConfig extends BaseExtension {
         customKey.convention(gradleProperty(CUSTOM_KEY_PROPERTY))
         customKeyFile.convention(gradleProperty(CUSTOM_KEY_FILE_PROPERTY)
             .map({ layout.projectDirectory.file(it) }))
-        directory.convention(repositoryDirectory)
+        directory.convention(repositoryDirectory.asFile.absolutePath)
     }
 
     @Input
     @Optional
     abstract Property<String> getType()
 
-    @InputDirectory
+    @Input
     @Optional
-    abstract DirectoryProperty getDirectory()
+    abstract Property<String> getDirectory()
 
     @Input
     @Optional

--- a/src/main/groovy/pl/allegro/tech/build/axion/release/infrastructure/config/ScmPropertiesFactory.groovy
+++ b/src/main/groovy/pl/allegro/tech/build/axion/release/infrastructure/config/ScmPropertiesFactory.groovy
@@ -8,7 +8,7 @@ class ScmPropertiesFactory {
     static ScmProperties create(VersionConfig config) {
         return new ScmProperties(
             config.repository.type.get(),
-            config.repository.directory.asFile.get(),
+            new File(config.repository.directory.get()),
             config.repository.remote.get(),
             config.repository.pushTagsOnly().get(),
             config.repository.fetchTags().get(),

--- a/src/test/groovy/pl/allegro/tech/build/axion/release/domain/RepositoryConfigFactoryTest.groovy
+++ b/src/test/groovy/pl/allegro/tech/build/axion/release/domain/RepositoryConfigFactoryTest.groovy
@@ -15,7 +15,7 @@ class RepositoryConfigFactoryTest extends Specification {
         RepositoryConfig config = Fixtures.repositoryConfig(project)
 
         then:
-        config.directory.asFile.get() == project.rootDir
+        new File(config.directory.get()) == project.rootDir
     }
 
     def "should not initialize authorization options when no flags on project"() {

